### DISCUSSION TITLE
WebGUI HTTPs Caddy custom SSL cert not recognized

### DISCUSSION THREAD

**Discussion by BcgA59SLMw** [postalserver/postal#3056] [Help with using Postal] [answered]
I am attempting to get caddy to use a cert I generated for the webgui. I have copied the certs to /opt/postal/config/ and my domain folder in /opt/postal/caddy-data/caddy/certificates/local/domain/

I edited the Caddy file and added the tls directive:
tls internal /config/fullchain.pem /config key.pem

I've verified the key is RSA as well. 

However, my webgui always uses the caddy generated cert and not the one I made. I am sure I'm missing something obvious - anything I'm missing or other places to look?

**Comment by willpower232** [comment]
How are you running caddy? It is likely not running inside the Postal container so you will probably need to provide the certificates another way to it, probably a different docker mount.

**Comment by BcgA59SLMw** [comment, marked as answer]
Thanks for the response. It was as always something pretty simple :) I'm using the default postal setup so /opt/caddy/config is mounted under /config for the web server. I did everything correctly (put the cert files in the ./config directory - but I had the order in my CaddyFile wrong. Needed to be:
myserver {
  reverse_proxy 127.0.0.1:5000
  tls internal /config/fullchain.pem /config/key.pem
}

And I had the tls line first before the reverse proxy line. Thanks again for your time. So now for anyone else who is wondering - I have the default postal setup, but with a cert I signed using our local CA and all is well. (You need to also configure your browser to accept local CAs in order to fully get rid of the error). 

Jonathan
